core/app/models/spree/order_update_attributes.rb

```diff
@@ -0,0 +1,36 @@
+module Spree
+  class OrderUpdateAttributes
+    attr_reader :attributes, :payments_attributes, :order, :request
+    def initialize(order, attributes, request_env: nil, request: nil)
+      @order = order
+      @attributes = attributes.dup
+      @payments_attributes = @attributes.delete(:payments_attributes) || []
+      @request = request
+      @request_env = request_env
+      @request_env ||= request.headers.env if request
+    end
+
+    def update
```

Comment:
What do you think of `apply` instead of `update` for the name of this method?  Or something else like that.  If I have a "FooAttributes" object, I kind expect `foo_attributes.update` to refresh the attributes themselves or something like that.